test_conformance/subgroups/test_workgroup_non_uniform_vote.cpp

```diff
@@ -0,0 +1,471 @@
+                        ? calculation_result[0]
+                        : calculation_result[1];
+                    if (device_result != expected_result)
+                    {
+                        log_error("ERROR: sub_group_non_uniform_%s mismatch "
+                                  "for local id %d in sub group %d in group "
+                                  "%d, obtained %d, expected %d\n",
+                                  operation_names(operation), i, j, k,
+                                  device_result, expected_result);
+                        return TEST_FAIL;
+                    }
+                }
+            }
+            x += nw;
+            y += nw;
+            m += 4 * nw;
+        }
+        log_info("  sub_group_non_uniform_%s(%s)... passed\n",
+                 operation_names(operation), TypeManager<Ty>::name());
+        return TEST_PASS;
+    }
+};
+
+// Test for elect function.
+// Discover only one elected work item in subgroup - with the
+// lowest subgroup local id
+struct ELECT
+{
+    static void gen(cl_int *x, cl_int *t, cl_int *m, int ns, int nw, int ng)
+    {
+        int non_uniform_size = ng % nw;
+        log_info("  sub_group_elect...\n");
+        if (non_uniform_size)
+        {
+            log_info("  non uniform work group size mode ON\n");
+        }
+        // no work here needed.
+    }
+
+    static int chk(cl_int *x, cl_int *y, cl_int *mx, cl_int *my, cl_int *m,
+                   int ns, int nw, int ng)
+    {
+        int ii, i, j, k, n;
+        int nj = (nw + ns - 1) / ns;
+        cl_int tr, rr;
+        int non_uniform_size = ng % nw;
+        ng = ng / nw;
+        if (non_uniform_size) ng++;
+        int last_subgroup_size = 0;
+
+
+        for (k = 0; k < ng; ++k)
+        { // for each work_group
+            if (non_uniform_size && k == ng - 1)
+            {
+                set_last_worgroup_params(non_uniform_size, nj, ns, nw,
+                                         last_subgroup_size);
+            }
+            for (j = 0; j < nw; ++j)
+            { // inside the work_group
+                i = m[4 * j + 1] * ns + m[4 * j];
+                my[i] = y[j]; // read device outputs for work_group
+            }
+
+            for (j = 0; j < nj; ++j)
+            { // for each subgroup
+                ii = j * ns;
+                if (last_subgroup_size && j == nj - 1)
+                {
+                    n = last_subgroup_size;
+                }
+                else
+                {
+                    n = ii + ns > nw ? nw - ii : ns;
+                }
+                rr = 0;
+                for (i = 0; i < n; ++i)
+                { // for each work_item in subgroup
+                  // sum of output values should be 1
+
+                    my[ii + i] > 0 ? rr += 1 : rr += 0;
+                }
+                // expectation is that only one elected returned true
+                tr = n <= NR_OF_ACTIVE_WORK_ITEMS ? 1 : 2;
```

Comment:
If we're expecting only one elected work item in the subgroup, why "2" here?

Reply:
fixed